diff -crN '--exclude=.git' '--exclude=.gitee' '--exclude=.vscode' lz4/lib/lz4.c lz4_h1/lib/lz4.c
*** lz4/lib/lz4.c	2022-08-26 15:00:24.524000000 +0800
--- lz4_h1/lib/lz4.c	2022-08-26 15:00:10.636000000 +0800
***************
*** 1749,1755 ****
                   const size_t dictSize         /* note : = 0 if noDict */
                   )
  {
!     if (src == NULL) { return -1; }
  
      {   const BYTE* ip = (const BYTE*) src;
          const BYTE* const iend = ip + srcSize;
--- 1749,1755 ----
                   const size_t dictSize         /* note : = 0 if noDict */
                   )
  {
!     if ((src == NULL) || (outputSize < 0)) { return -1; }
  
      {   const BYTE* ip = (const BYTE*) src;
          const BYTE* const iend = ip + srcSize;
diff -crN '--exclude=.git' '--exclude=.gitee' '--exclude=.vscode' lz4/lib/lz4hc.c lz4_h1/lib/lz4hc.c
*** lz4/lib/lz4hc.c	2022-08-26 15:00:24.524000000 +0800
--- lz4_h1/lib/lz4hc.c	2022-08-26 15:00:10.640000000 +0800
***************
*** 955,960 ****
--- 955,961 ----
  {
  #if defined(LZ4HC_HEAPMODE) && LZ4HC_HEAPMODE==1
      LZ4_streamHC_t* const statePtr = (LZ4_streamHC_t*)ALLOC(sizeof(LZ4_streamHC_t));
+     if (statePtr == NULL) return 0;
  #else
      LZ4_streamHC_t state;
      LZ4_streamHC_t* const statePtr = &state;
